src/Products.jl

```diff
@@ -494,6 +494,7 @@ function locate(fp::FileProduct, prefix::Prefix; platform::AbstractPlatform = Ho
 end
 
 # Necessary to get the products in the wrappers always sorted consistently
+Base.isless(x::ExecutableProduct, y::LibraryProduct) = true
```

Comment:
If anything, I'd put libraries before executables: they're usually more fundamental, executables typically depend on libraries, the opposite doesn't usually make sense

Reply:
That makes sense, I was just going with a "E comes before L" thought.
I thought just flipping the method arguments would do but only adding the `true` and 'false' case resulted in the correct ordering.

```
#Base.isless(x::LibraryProduct, y::ExecutableProduct) = false

9-element Vector{Product}:
 ExecutableProduct(["dltest"], :dltest, nothing)
 ExecutableProduct(["isql"], :isql, nothing)
 ExecutableProduct(["iusql"], :iusql, nothing)
 LibraryProduct(["libodbc"], :libodbc, String[], false, Symbol[])
 ExecutableProduct(["odbc_config"], :odbc_config, nothing)
 LibraryProduct(["libodbcinst"], :libodbcinst, String[], false, Symbol[])
 ExecutableProduct(["odbcinst"], :odbcinst, nothing)
 LibraryProduct(["libodbccr"], :libodbccr, String[], false, Symbol[])
 ExecutableProduct(["slencheck"], :slencheck, nothing)


#Base.isless(x::LibraryProduct, y::ExecutableProduct) = true
 julia> sort(products)
9-element Vector{Product}:
 ExecutableProduct(["dltest"], :dltest, nothing)
 ExecutableProduct(["isql"], :isql, nothing)
 ExecutableProduct(["iusql"], :iusql, nothing)
 LibraryProduct(["libodbc"], :libodbc, String[], false, Symbol[])
 LibraryProduct(["libodbccr"], :libodbccr, String[], false, Symbol[])
 LibraryProduct(["libodbcinst"], :libodbcinst, String[], false, Symbol[])
 ExecutableProduct(["odbc_config"], :odbc_config, nothing)
 ExecutableProduct(["odbcinst"], :odbcinst, nothing)
 ExecutableProduct(["slencheck"], :slencheck, nothing)

#Base.isless(x::ExecutableProduct, y::LibraryProduct) = false

julia> sort(products)
9-element Vector{Product}:
 LibraryProduct(["libodbc"], :libodbc, String[], false, Symbol[])
 LibraryProduct(["libodbcinst"], :libodbcinst, String[], false, Symbol[])
 ExecutableProduct(["isql"], :isql, nothing)
 ExecutableProduct(["iusql"], :iusql, nothing)
 LibraryProduct(["libodbccr"], :libodbccr, String[], false, Symbol[])
 ExecutableProduct(["dltest"], :dltest, nothing)
 ExecutableProduct(["odbc_config"], :odbc_config, nothing)
 ExecutableProduct(["odbcinst"], :odbcinst, nothing)
 ExecutableProduct(["slencheck"], :slencheck, nothing)

#Base.isless(x::LibraryProduct, y::ExecutableProduct) = true
#Base.isless(x::ExecutableProduct, y::LibraryProduct) = false

 9-element Vector{Product}:
 LibraryProduct(["libodbc"], :libodbc, String[], false, Symbol[])
 LibraryProduct(["libodbccr"], :libodbccr, String[], false, Symbol[])
 LibraryProduct(["libodbcinst"], :libodbcinst, String[], false, Symbol[])
 ExecutableProduct(["dltest"], :dltest, nothing)
 ExecutableProduct(["isql"], :isql, nothing)
 ExecutableProduct(["iusql"], :iusql, nothing)
 ExecutableProduct(["odbc_config"], :odbc_config, nothing)
 ExecutableProduct(["odbcinst"], :odbcinst, nothing)
 ExecutableProduct(["slencheck"], :slencheck, nothing)

```